calculator.rb

```diff
@@ -0,0 +1,99 @@
+# This is a calculator program
+# the user should ask what they would like to calculate.
+
+operations = ["multiply", "divide", "subtract", "exponify",
+"modulate", "-", "*", "/", "**", "+", "%"]
+
+puts "\nWelcome to your Calculator Program!"
+puts "Enter two numbers and an operation."
+puts "Would you like to continue? (yes/no) "
+answer = gets.chomp.downcase
+
+while answer == "yes" do
+  # first number
+  print "\n#1 Number: "
+  until (num_1 = gets.chomp) =~ /^\d*\.?\d+$/ 
+#starting of line, any digit(zero or more), including periods, following any digit(zero or more), ending of string.
+    puts "\nInvalid input. Please enter a number: "
+  end
+  num_1 = num_1.to_f
+  # second number
+  print "\n#2 Number: "
+  until (num_2 = gets.chomp) =~ /^\d*\.?\d+$/
+    puts "\nInvalid input. Please enter a number: "
+  end
+  num_2 = num_2.to_f
+  # operation to perform
+  print "\nOperation: "
+  op = gets.chomp.downcase
+  until operations.include?("#{op}")
+    print "\nPlease enter an operation
+    (add, subtract, multiply, divide, exponify, or modulate): "
+    op = gets.chomp.downcase
+  end
+
+  # divide
+  def divide(num1, num2)
+    if num2 != 0
+
```

Comment:
This `divide` method is defined _inside_ your while loop! While this is technically valid Ruby (and certainly works), it's a little odd to me as a reader. A better option might be to define all your methods at the very top of the file, and have your main control loop below that.